### ISSUE TITLE
Syntax error:

### ISSUE THREAD

**Issue by Riftcore34** [ewhal/nyaa#183]
/go# go get github.com/ewhal/nyaa
# github.com/mattn/go-sqlite3
exec: "gcc": executable file not found in $PATH
# github.com/jinzhu/gorm
src/github.com/jinzhu/gorm/utils.go:143: syntax error: unexpected range, expecting {
src/github.com/jinzhu/gorm/utils.go:153: non-declaration statement outside function body
src/github.com/jinzhu/gorm/utils.go:154: syntax error: unexpected }

**Comment by majestrate** [comment]
not our bug.
(this is why vendoring deps is a win btw)

**Comment by Riftcore34** [comment]
Didn't mean to set it as a bug, Was asking for help.

**Comment by majestrate** [comment]
looks like gorm broke their code possibly, you'll need gcc too it seems. idk if this project builds on win32 rn...

**Comment by Riftcore34** [comment]
Sorry this is on debian should of put that in post one.

**Comment by sfan5** [comment]
go version?

**Comment by Riftcore34** [comment]
go version go1.3.3 linux/amd64

**Comment by Leodmanx2** [comment]
> idk if this project builds on win32 rn...
It does, using Cygwin or equivalent. GCC only.

> go version go1.3.3 linux/amd64
That's pretty out-of-date. Presumably https://golang.org/doc/go1.4#language explains your issue.

**Comment by Riftcore34** [comment]
out of date? I just did a fresh install (apt-get install Golang) I have no clue about go never even heard of it. So this is my end?

**Comment by Atvaark** [comment]
Try downloading go 1.8

`apt-get install golang-1.8-go`

**Comment by Riftcore34** [comment]
apt-get install golang-1.8-go
Reading package lists... Done
Building dependency tree
Reading state information... Done
E: Unable to locate package golang-1.8-go
E: Couldn't find any package by regex 'golang-1.8-go'

**Comment by Riftcore34** [comment]
apt-get install golang
Reading package lists... Done
Building dependency tree
Reading state information... Done
golang is already the newest version.

Its odd because it's telling me im upto date.

```
apt-get install golang
Reading package lists... Done
Building dependency tree
Reading state information... Done
golang is already the newest version.
```

**Comment by Leodmanx2** [comment]
That package is only available in the testing and unstable branches of Debian. You're using the stable branch (Debian 8/Jessie) it seems.

See https://packages.debian.org/search?suite=default&section=all&arch=any&searchon=names&keywords=golang

**Comment by Riftcore34** [comment]
Seems like this is bloody hard lol. What OS would you say is best just for this Git. I only want to run this on a spare computer.

**Comment by kipukun** [comment]
Ubuntu 16.10 would be the easiest OS to use.
Run `sudo apt-get install golang-1.8-go` once you have Ubuntu installed.
Run `go get github.com/ewhal/nyaa` in any directory.

I wouldn't recommend trying to pull this repo yet, though. Everything is constantly being broken and fixed, so unless you're in the dev IRC channel, it's a pain to keep up with.

**Comment by Leodmanx2** [comment]
I'd also recommend Ubuntu.
I remember @ewhal saying 1.8 was required.
@kipukun recommends 16.10, but searching the package online only shows the 1.8 package as being available in 17.04 and up. Thankfully, that's the current release.

**Comment by kipukun** [comment]
Woops, forgot Ubuntu upped their version a month ago. Apologies.
Yeah then in that case, 17.04.
